Core/include/Acts/Surfaces/detail/IntersectionHelper2D.hpp

```diff
@@ -0,0 +1,56 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <utility>
+
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Intersection.hpp"
+
+namespace Acts {
+namespace detail {
+
+struct IntersectionHelper2D {
+  /// Intersect two segments
+  ///
+  /// @param s0 The Start of the segement
+  /// @param s1 The end of the segement
+  /// @param origin The Start of intersection line
+  /// @param direction The Direction of intersection line
```

Comment:
Maybe you should add a return line similar to the one for ellipses and circle ?

Reply:
Fixed.